onnxruntime/test/providers/cpu/math/matmul_integer_test.cc

```diff
@@ -30,5 +30,54 @@ TEST(MatmulIntegerOpTest, MatMulInteger) {
   test.AddOutput<int32_t>("T3", {1, 1}, {-1});
   test.Run();
 }
+
+TEST(MatmulIntegerOpTest, MatMulInteger_Uint8_Int8_GEMM) {
+  OpTester test("MatMulInteger", 10);
+  test.AddInput<uint8_t>("T1", {2, 40}, {4, 26, 2, 69, 76, 14, 14, 44, 58, 2, 21, 21, 60, 32, 59, 23, 74, 56, 37, 25, 26, 31, 60, 37, 46, 18, 30, 56, 3, 38, 55, 72, 31, 56, 37, 52, 11, 50, 57, 74, 48, 59, 33, 0, 51, 22, 73, 50, 70, 43, 34, 63, 74, 63, 34, 15, 43, 36, 52, 72, 7, 17, 33, 49, 3, 67, 47, 12, 75, 62, 43, 46, 43, 3, 18, 8, 61, 19, 58, 13});
+  test.AddInput<int8_t>("T2", {40, 2}, {12, 10, 9, 13, 2, 7, 10, 4, 17, 15, 14, 19, 4, 11, 1, 19, 7, 0, 12, 3, 13, 8, 4, 12, 16, 18, 15, 12, 9, 8, 12, 19, 14, 11, 18, 15, 9, 3, 0, 13, 5, 1, 5, 3, 8, 3, 2, 17, 0, 0, 4, 13, 6, 12, 11, 17, 9, 10, 5, 8, 15, 5, 4, 18, 5, 0, 4, 10, 11, 4, 4, 0, 9, 14, 14, 17, 19, 13, 3, 13});
+  test.AddOutput<int32_t>("T3", {2, 2}, {14110, 15747, 14114, 17441});
+  test.Run();
+}
+
+TEST(MatmulIntegerOpTest, MatMulInteger_Uint8_Int8_GEMV_1) {
+  OpTester test("MatMulInteger", 10);
+  test.AddInput<uint8_t>("T1", {1, 64}, {1, 9, 0, 20, 24, 22, 22, 18, 20, 2, 21, 13, 9, 3, 19, 8, 16, 16, 9, 15, 2, 21, 4, 18, 2, 2, 10, 16, 0, 19, 17, 13, 21, 21, 7, 15, 5, 8, 23, 14, 17, 0, 21, 23, 24, 7, 18, 9, 17, 13, 14, 23, 1, 18, 3, 17, 5, 17, 15, 24, 24, 9, 22, 7});
+  test.AddInput<int8_t>("T2", {64, 2}, {18, 16, 3, 13, 14, 9, 18, 17, 7, 12, 11, 5, 11, 0, 13, 13, 9, 17, 10, 12, 10, 3, 16, 12, 5, 10, 2, 2, 3, 15, 17, 3, 0, 3, 16, 6, 19, 19, 17, 2, 5, 7, 11, 19, 2, 0, 15, 11, 0, 8, 17, 10, 1, 11, 8, 14, 8, 13, 7, 13, 11, 13, 5, 18, 4, 3, 16, 12, 10, 12, 4, 10, 11, 18, 19, 3, 5, 4, 9, 2, 19, 4, 17, 6, 5, 6, 14, 3, 18, 3, 13, 17, 16, 2, 5, 2, 15, 4, 16, 14, 11, 18, 0, 17, 1, 17, 8, 3, 18, 13, 7, 1, 7, 6, 11, 8, 7, 11, 10, 13, 16, 16, 4, 19, 4, 9, 4, 14});
+  test.AddOutput<int32_t>("T3", {1, 2}, {8553, 7926});
+  test.Run();
+}
+
+TEST(MatmulIntegerOpTest, MatMulInteger_Uint8_Int8_1) {
```

Comment:
How is MatMulInteger_Uint8_Int8_1, MatMulInteger_Uint8_Int8_2 and MatMulInteger_Uint8_Int8_3 different?

Reply:
The shape is the main difference. Will update with random generated data to make it clear.

---
In reply to: [303683326](https://github.com/microsoft/onnxruntime/pull/1391#discussion_r303683326) [](ancestors = 303683326)